### PR TITLE
made the text and link for the twitter share dynamic

### PR DESCRIPTION
**Pull request by jhammond2012**
image

image

In this PR, both the link and text are dynamic.

### CODE REVIEW

**Comment by encody** [comment]
Part of the issue is that the Twitter button code actually *replaces* the HTML in the page with an `<iframe>`, so the computed properties don't update on page navigation because the target element no longer exists (i.e. it has been replaced with Twitter's custom code). One solution to this problem would be to replace [this code](https://github.com/NEAR-Edu/stats.gallery/blob/main/src/views/Page.vue#L23-L33) with a dynamically *re-generated* bit of HTML code that we stick into a [`v-html`](https://v3.vuejs.org/guide/template-syntax.html#raw-html) binding, and then [re-run the Twitter button replacer code](https://github.com/NEAR-Edu/stats.gallery/blob/main/src/App.vue#L115-L122).

**Comment by jhammond2012** [comment]
@encody I have updated this to rerender on route change.

File: src/views/Page.vue [hunk 1]
```diff
@@ -343,9 +335,12 @@ export default defineComponent({
     });
 
     const shareRoute = computed(() => {
+      console.log('route', window.location);
       let shareLink = 'https://twitter.com/share?ref_src=twsrc%5Etfw';
 
-      shareLink = `https://twitter.com/share?ref_src=${route}`;
+      shareLink = `https://twitter.com/share?ref_src=${
```
Comment (encody) [inline src/views/Page.vue:341]: Could this be simplified to a single `return` statement?

File: src/views/Page.vue [hunk 2]
```diff
@@ -371,7 +366,28 @@ export default defineComponent({
       NftBadge,
       tweetShareText,
       shareRoute,
+      tweetButton: '',
```
Comment (encody) [inline src/views/Page.vue:369]: Prefer `ref` (composition API)

File: src/views/Page.vue [hunk 3]
```diff
@@ -371,7 +366,28 @@ export default defineComponent({
...
+        data-text="${this.tweetShareText}"
+        href="${this.shareRoute}">
+      Tweet</a>`;
+    },
+  },
+  watch: {
+    shareRoute() {
+      this.renderTweetButton();
+    },
+  },
```
Comment (encody) [inline src/views/Page.vue:391]: Prefer composition API

**Review by encody — CHANGES_REQUESTED** [review]

**Comment by encody** [comment]
The Tweet button does not appear when loading the overview page from the landing page.

File: src/views/Page.vue [hunk 4]
```diff
@@ -327,6 +319,32 @@ export default defineComponent({
...
+      let tweetText = 'Check out my NEAR stats!';
+
+      const routeArr = route.path.split('/');
+      const routeSlug = routeArr[routeArr.length - 1];
+      if (routeSlug == 'stats') tweetText = 'Check out my NEAR stats!';
+      if (routeSlug == 'leaderboards')
+        tweetText = 'Check out the leaderboards on NEAR!';
+      if (routeSlug == 'transactions')
+        tweetText = 'Check out my transactions on NEAR!';
+      if (routeSlug == 'quests') tweetText = 'Check out my quests on NEAR!';
```
Comment (encody) [inline src/views/Page.vue:332]: Could this be a switch/case, or even an object that maps route slugs to messages? Might be a little more readable that way.

File: src/views/Page.vue [hunk 5]
```diff
@@ -327,6 +319,32 @@ export default defineComponent({
 
     const route = useRoute();
 
+    const tweetShareText = computed(() => {
+      let tweetText = 'Check out my NEAR stats!';
+
+      const routeArr = route.path.split('/');
+      const routeSlug = routeArr[routeArr.length - 1];
```
Comment (encody) [inline src/views/Page.vue:326]: Instead of route slug, you might be able to use the route's name.

**Review by jhammond2012 — COMMENTED** [review]
Changes have been made and committed to this branch.

**Review by encody — APPROVED** [review]

**Pull request merged**